src/test/java/com/rabbitmq/jms/admin/RMQObjectFactoryTest.java

```diff
@@ -0,0 +1,165 @@
+package com.rabbitmq.jms.admin;
+
+import org.junit.Test;
+
+import javax.jms.ConnectionFactory;
+import javax.naming.CompositeName;
+import javax.naming.NamingException;
+import javax.naming.Reference;
+import java.util.Hashtable;
+
+import static org.junit.Assert.*;
+
+/**
+ * RMQObjectFactory unit test
+ *
+ * @author Rémi Bantos
+ */
+public class RMQObjectFactoryTest {
+
+    private RMQObjectFactory rmqObjectFactory = new RMQObjectFactory();
+
+    @Test
+    public void getObjectInstance_should_create_a_RMQConnectionFactory_via_Reference() throws Exception {
+
+        Reference ref = new Reference(ConnectionFactory.class.getName());
+
+        Object createdObject = rmqObjectFactory.getObjectInstance(ref, new CompositeName("java:global/jms/TestConnectionFactory"), null, null);
+
+        assertNotNull(createdObject);
+        assertEquals(RMQConnectionFactory.class, createdObject.getClass());
+
+        RMQConnectionFactory createdConFactory = (RMQConnectionFactory) createdObject;
+
+        assertEquals("guest", createdConFactory.getUsername());
+        assertEquals("guest", createdConFactory.getPassword());
+        assertEquals("/", createdConFactory.getVirtualHost());
+        assertEquals("localhost", createdConFactory.getHost());
+
+    }
+
+
+    @Test
+    public void getObjectInstance_should_create_a_RMQConnectionFactory_via_environment() throws Exception {
```

Comment:
I use to mix both especially for readability of test cases. But indeed, it seems that we use CamelCase everywhere in this project. I'm ok to change that.